apps/meteor/client/views/room/providers/SelectedMessagesProvider.tsx

```diff
@@ -61,13 +84,24 @@ export const selectedMessageStore = new (class SelectMessageStore extends Emitte
 		this.isSelecting = false;
 		this.emit('toggleIsSelecting', false);
 	}
+
+	toggleAll(mids: string[]): void {
+		mids.forEach((mid) => this.select(mid));
+	}
```

Comment:
Instead of a foreach, is it possible to have a method in this class that does a bulk procedure? This way we avoid emmiting a ton of events from the store.

```
this.store = new Set([...this.store, ...mids])
```

I think this should work. A foreach can also work, but then you should emit the 'change' event only once after the operation has ended, instead of one time for each entry.